neqo-transport/src/crypto.rs

```diff
@@ -1490,38 +1493,60 @@ impl CryptoStreams {
         &mut self,
         space: PacketNumberSpace,
         builder: &mut PacketBuilder,
+        shuffle: bool,
         tokens: &mut Vec<RecoveryToken>,
         stats: &mut FrameStats,
     ) {
-        let cs = self.get_mut(space).unwrap();
-        if let Some((offset, data)) = cs.tx.next_bytes() {
-            let mut header_len = 1 + Encoder::varint_len(offset) + 1;
+        fn write_chunks(
+            chunks: Vec<(u64, &[u8])>,
+            builder: &mut PacketBuilder,
+        ) -> Vec<(u64, usize)> {
+            let mut written = vec![];
+            for (offset, data) in chunks {
+                let mut header_len = 1 + Encoder::varint_len(offset) + 1;
+
+                // Don't bother if there isn't room for the header and some data.
+                if builder.remaining() < header_len + 1 {
+                    break;
+                }
+                // Calculate length of data based on the minimum of:
+                // - available data
+                // - remaining space, less the header, which counts only one byte for the length at
+                //   first to avoid underestimating length
+                let length = min(data.len(), builder.remaining() - header_len);
+                header_len += Encoder::varint_len(u64::try_from(length).unwrap()) - 1;
+                let length = min(data.len(), builder.remaining() - header_len);
+
+                builder.encode_varint(crate::frame::FRAME_TYPE_CRYPTO);
+                builder.encode_varint(offset);
+                builder.encode_vvec(&data[..length]);
+                written.push((offset, length));
+            }
+            written
+        }
 
-            // Don't bother if there isn't room for the header and some data.
-            if builder.remaining() < header_len + 1 {
-                return;
+        let cs: &mut CryptoStream = self.get_mut(space).unwrap();
+        if let Some((offset, data)) = cs.tx.next_bytes() {
+            let chunks = if shuffle {
+                // Mix up the crypto data a bit. Create five chunks that split ASCII LDHD sequences,
+                // and reorder them.
```

Comment:
Five is no longer the case

```suggestion
                // Cut the crypto data in two at random and swap the chunks.
```